test/functional/plugin/lsp_spec.lua

```diff
@@ -1058,8 +1058,7 @@ describe('LSP', function()
       local client
       test_rpc_server {
         test_name = 'basic_finish',
-        on_init = function(_client)
-          client = _client
```

Comment:
Again attaching a client to a buffer in `on_init` (which happens after it has initialized) should cause it to call `didOpen`.